pkg/controller/common/test/runner.go

```diff
@@ -18,78 +18,72 @@ import (
-			t.Helper()
-			utilruntime.ErrorHandlers = append(utilruntime.ErrorHandlers, func(err error) { t.Errorf("unhandled error occurred in k8s: %v", err) })
-			defer func() {
-				// XXX: this is pretty sketchy...
-				utilruntime.ErrorHandlers = utilruntime.ErrorHandlers[:len(utilruntime.ErrorHandlers)-1]
-			}()
-			if testCase.ConfigureGlobals != nil {
-				testCase.ConfigureGlobals()
-			}
-			mgr, tracker, err := NewManagerForControllerTest(testCase.GroupResources...)
-			if err != nil {
-				t.Fatal(err)
-			}
-			for _, resource := range testCase.Resources {
-				// XXX: should we use client.Create() or tracker.Add()?
-				// client.Create() has side effects, like adding creation time, resource version, etc.
-				if err := mgr.GetClient().Create(context.TODO(), resource); err != nil {
-					t.Fatal(err)
+	utilruntime.ErrorHandlers = append(utilruntime.ErrorHandlers, func(err error) { t.Errorf("unhandled error occurred in k8s: %v", err) })
+	defer func() {
+		// XXX: this is pretty sketchy...
+		utilruntime.ErrorHandlers = utilruntime.ErrorHandlers[:len(utilruntime.ErrorHandlers)-1]
+	}()
+	if testCase.ConfigureGlobals != nil {
+		testCase.ConfigureGlobals()
+	}
+	mgr, tracker, err := NewManagerForControllerTest(testCase.GroupResources...)
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, resource := range testCase.Resources {
+		// XXX: should we use client.Create() or tracker.Add()?
+		// client.Create() has side effects, like adding creation time, resource version, etc.
+		if err := mgr.GetClient().Create(context.TODO(), resource); err != nil {
+			t.Fatal(err)
+		}
+	}
+	for _, addController := range testCase.AddControllers {
+		if err := addController(mgr); err != nil {
+			t.Fatal(err)
+		}
+	}
+	stop := StartManager(mgr, t)
+	func() {
+		t.Helper()
+		defer stop()
+		for _, event := range testCase.Events {
+			t.Run(event.Name, func(t *testing.T) {
+				defer func() {
+					tracker.RemoveReaction(event.Verifier)
+					for _, assertion := range event.Assertions {
+						tracker.RemoveReaction(assertion)
+					}
+					tracker.RemoveReaction(event.Reactors...)
+				}()
+				// inject the test runner into the verifier
+				event.Verifier.InjectTestRunner(t)
+				// insert reactions.  these must come before any default reactions added for normal resource handling
+				tracker.PrependReaction(event.Reactors...)
+				// insert assertions.  these need to be before any reactors, as they do not actually handle events
+				for _, assertion := range event.Assertions {
+					tracker.PrependReaction(assertion)
 				}
-			}
-			for _, addController := range testCase.AddControllers {
-				if err := addController(mgr); err != nil {
+				// insert verifier.  this needs to be the first handler, as it verifies the event, but does not handle it
+				tracker.PrependReaction(event.Verifier)
+				// add failure for events occurring after validation should be complete
+				tracker.PrependReaction(&extraneousActionFailure{verifier: event.Verifier, t: t})
```

Comment:
This seems to be causing the failure

Reply:
The error seems to imply another event is coming through the reconcile loop, hence the "get" request for an "smcp" resource named, "test."  I suspect this has to do with the `reflect.DeepEqual()` call in `skipStatusUpdate()`.  I wonder if this is the result of different go versions.  (I needed to modify the `LastTransitionTime` initialization, as `DeepCopy()`ing the object caused the values to be different.)

TBH, I suspect using `reflect.DeepEqual()` might be a little flaky for this use case.

Reply:
See comments below, but this is working as desired now.